File: core/src/com/unibo/s3/main_system/rendering/ScaleUtils.java
package com.unibo.s3.main_system.rendering;

/**
 *
 * @author mvenditto
 * */
public final class ScaleUtils {

    private static final int PIXELS_PER_METER = 50;
    private static final float METERS_PER_PIXEL = 1.0f / PIXELS_PER_METER;

    public static float getMetersPerPixel() {
        return METERS_PER_PIXEL;
    }

    public static int getPixelsPerMeter() {
        return PIXELS_PER_METER;
    }

    public static float pixelsToMeters (int pixels) {
        return (float)pixels * METERS_PER_PIXEL;
    }

    public static int metersToPixels (float meters) {
        return (int)(meters * PIXELS_PER_METER);
    }

}

File: core/src/com/unibo/s3/main_system/util/ScaleUtils.scala
package com.unibo.s3.main_system.util

/**
 *
 * @author mvenditto
 * */
object ScaleUtils {

    private[this] val PIXELS_PER_METER: Integer = 50
    private[this] val METERS_PER_PIXEL: Float = 1.0f / PIXELS_PER_METER

    def getMetersPerPixel: Float = METERS_PER_PIXEL

    def getPixelsPerMeter: Integer = PIXELS_PER_METER

    def pixelsToMeters(pixels: Integer): Float = pixels * METERS_PER_PIXEL

    def metersToPixels(meters: Float): Integer = (meters * PIXELS_PER_METER).toInt

}
